textattack/metrics/metrics.py

```diff
@@ -0,0 +1,161 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from textattack.attack_results import (
+    FailedAttackResult,
+    SkippedAttackResult,
+    SuccessfulAttackResult,
+)
+
+
+class Metric(ABC):
+    """A metric processes a list of `AttackResult` objects for a given
+    statistic."""
+
+    @property
+    def key(self):
+        """The string representation of the metric, like 'Attack Success Rate'
+        or 'Levenshtein Distance'."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def compute(attack_results):
+        """Returns a string representation of a metric calculated across a list
+        of attack results."""
+        raise NotImplementedError()
+
+
+class TotalAttacks(Metric):
+    key = "Number of attacks"
+
+    @staticmethod
+    def compute(attack_results):
+        total = sum((1 for r in attack_results))
+        return str(total)
+
+
+class TotalSuccessfulAttacks(Metric):
+    key = "Successful attacks"
+
+    @staticmethod
+    def compute(attack_results):
+        return sum((isinstance(r, SuccessfulAttackResult) for r in attack_results))
+
+
+class TotalFailedAttacks(Metric):
+    key = "Failed attacks"
+
+    @staticmethod
+    def compute(attack_results):
+        total = sum((isinstance(r, FailedAttackResult) for r in attack_results))
+        return total
+
+
+class TotalSkippedAttacks(Metric):
+    key = "Skipped attacks"
+
+    @staticmethod
+    def compute(attack_results):
+        total = sum((isinstance(r, SkippedAttackResult) for r in attack_results))
+        return total
+
+
+class ModelAccuracy(Metric):
+    key = "Model accuracy"
+
+    @staticmethod
+    def compute(attack_results):
+        total_attacks = len(attack_results)
+        skipped_attacks = TotalSkippedAttacks.compute(attack_results)
+        model_accuracy = (total_attacks - skipped_attacks) * 100.0 / (total_attacks)
```

Comment:
Not introduced by your changes but this actually seems like an unreliable way of computing accuracy, e.g. for targeted classification even if the model is wrong the goal might not be met yet if there are more than 2 classes, so the attack won't be skipped even though it's wrong.

Reply:
Actually this also applied for `AccuracyUnderAttack` to some extent, perhaps we should say that these only make sense for untargeted classification attacks.